pkg/alertmanager/multitenant.go

```diff
@@ -1011,6 +1012,102 @@ func (am *MultitenantAlertmanager) ReplicateStateForUser(ctx context.Context, us
 	return err
 }
 
+// getReplicasForUser returns the addresses for each _other_ replica assigned to a user.
+func (am *MultitenantAlertmanager) getOtherReplicasForUser(userID string) ([]string, error) {
+	if am.ring.InstancesCount() <= 0 {
+		return nil, fmt.Errorf("InstancesCount <= 0")
+	}
+
+	key := shardByUser(userID)
+	replicationSet, err := am.ring.Get(key, RingOp, nil, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Don't return our own address in the list.
+	selfAddress := am.ringLifecycler.GetInstanceAddr()
+
+	result := make([]string, 0, len(replicationSet.Instances))
+	for _, desc := range replicationSet.Instances {
+		addr := desc.GetAddr()
+		if addr != selfAddress {
+			result = append(result, addr)
+		}
+	}
+
+	return result, nil
+}
+
+// ReadFullSateForUser attempts to read the full state from each replica for user. Note that it will try to obtain and return
+// state from all replicas, but will consider it a success if state is obtained from only a single replica.
```

Comment:
```suggestion
// state from all replicas, but will consider it a success if state is obtained from at least a replica.
```

Reply:
I see the mistake, but changed to `at least one` replica instead.